hydra-node/exe/hydra-node/Main.hs

```diff
@@ -31,23 +33,48 @@ import Hydra.Node (
   initEnvironment,
   runHydraNode,
  )
-import Hydra.Options (ChainConfig (..), LedgerConfig (..), Options (..), parseHydraOptions)
+import Hydra.Options (
+  ChainConfig (..),
+  Command (Publish, Run),
+  LedgerConfig (..),
+  Options (..),
+  PublishOptions (..),
+  parseHydraCommand,
+ )
 import Hydra.Party (Party)
 
 main :: IO ()
 main = do
-  o@Options{verbosity, host, port, peers, apiHost, apiPort, monitoringPort, chainConfig, ledgerConfig} <- identifyNode <$> parseHydraOptions
-  env@Environment{party} <- initEnvironment o
-  withTracer verbosity $ \tracer' ->
-    withMonitoring monitoringPort tracer' $ \tracer -> do
-      eq <- createEventQueue
-      withChain tracer party (putEvent eq . OnChainEvent) chainConfig $ \oc ->
-        withNetwork (contramap Network tracer) host port peers (putEvent eq . NetworkEvent) $ \hn ->
-          withAPIServer apiHost apiPort party (contramap APIServer tracer) (putEvent eq . ClientEvent) $ \server -> do
-            withCardanoLedger ledgerConfig $ \ledger -> do
-              node <- createHydraNode eq hn ledger oc server env
-              runHydraNode (contramap Node tracer) node
+  command <- parseHydraCommand
+  case command of
+    Run options ->
+      run (identifyNode options)
+    Publish options ->
+      publish options
  where
+  run opts = do
+    let Options{verbosity, monitoringPort} = opts
+    env@Environment{party} <- initEnvironment opts
+    withTracer verbosity $ \tracer' ->
+      withMonitoring monitoringPort tracer' $ \tracer -> do
+        eq <- createEventQueue
+        let Options{hydraScriptsTxId, chainConfig} = opts
+        withChain tracer party (putEvent eq . OnChainEvent) hydraScriptsTxId chainConfig $ \oc -> do
+          let Options{host, port, peers} = opts
+          withNetwork (contramap Network tracer) host port peers (putEvent eq . NetworkEvent) $ \hn -> do
+            let Options{apiHost, apiPort} = opts
+            withAPIServer apiHost apiPort party (contramap APIServer tracer) (putEvent eq . ClientEvent) $ \server -> do
+              let Options{ledgerConfig} = opts
+              withCardanoLedger ledgerConfig $ \ledger -> do
+                node <- createHydraNode eq hn ledger oc server env
+                runHydraNode (contramap Node tracer) node
+
+  publish opts = do
+    (_, sk) <- readKeyPair (publishSigningKey opts)
+    let PublishOptions{publishNetworkId = networkId, publishNodeSocket = nodeSocket} = opts
+    txId <- publishHydraScripts networkId nodeSocket sk
+    putStrLn (decodeUtf8 (serialiseToRawBytesHex txId))
```

Comment:
We could do more fancy / structured here, like JSON or YAML. Especially if we end up refactoring to return the full registry as a map from output-references to script hashes.

Reply:
There is `serialiseToRawBytesHexText` which would allow usage of `putTextLn` if you want.

Reply:
That'd be the same result though 😅